specs/interop/liquidity_migration.md

```diff
@@ -0,0 +1,358 @@
+
+#### `convertToSuper`
+
+Converts `_amount` of legacy tokens with address `_legacyAddr`
+to the same `_amount` of `SuperchainERC20` tokens in address `_superAddr`.
+
+```solidity
+convertToSuper(address _legacyAddr, address _superAddr, uint256 _amount)
+```
+
+The function will
+
+1. Check that `_legacyAddr` and `_superAddr` are valid and paired.
+2. Check both tokens have the same amount of decimals.
+3. Burn `_amount` of `_legacyAddr` from `msg.sender`.
+4. Mint `_amount` of `_superAddr` to `msg.sender`.
+
+An example implementation that depends on the `checkPair()`
+access control function looks like this:
+
+```solidity
+function convertToSuper(address _legacyAddr, address _superAddr, uint256 _amount) public {
+  require(checkPair(_legacyAddr, _superAddr), "Invalid address pair");
+  require(IERC20(_legacyAddr).decimals() == IERC20(_superAddr).decimals(), "Decimals do not match")
+
+  IERC20(_legacyAddr).burn(msg.sender, _amount);
+  IERC20(_superAddr).mint(msg.sender, _amount);
+
+  emit ConvertedToSuper(_legacyAddr, _superAddr, msg.sender, _amount);
+}
+```
+
+#### `convertFromSuper`
+
+Converts `_amount` of `SuperchainERC20` tokens in address `_superAddr`
+to the same `_amount` of legacy tokens with address `_legacyAddr`.
+
+```solidity
+convertToSuper(address _legacyAddr, address _superAddr, uint256 _amount)
+```
+
+The function will
+
+1. Check that `_legacyAddr` and `_superAddr` are valid and paired.
+2. Check both tokens have the same amount of decimals.
+3. Burn `_amount` of `_superAddr` from `msg.sender`.
+4. Mint `_amount` of `_legacyAddr` to `msg.sender`.
+
+An example implementation that depends on the `checkPair()`
+access control function looks like this:
+
+```solidity
+function convertFromSuper(address _legacyAddr, address _superAddr, uint256 _amount) public {
+  require(checkPair(_legacyAddr, _superAddr), "Invalid address pair");
+  require(IERC20(_legacyAddr).decimals() == IERC20(_superAddr).decimals(), "Decimals do not match")
+
+  IERC20(_superAddr).burn(msg.sender, _amount);
+  IERC20(_legacyAddr).mint(msg.sender, _amount);
+
+  emit ConvertedFromSuper(_legacyAddr, _superAddr, msg.sender, _amount);
+}
+```
+
+### Events
+
+#### `ConvertedToSuper`
+
+It MUST trigger when anyone converts legacy tokens
+to `SuperchainERC20` with `convertToSuper`.
+
+```solidity
+event ConvertedToSuper(address indexed legacyAddr, address indexed superAddr, address indexed caller, uint256 amount, uint256 adjustedAmount);
+```
+
+#### `ConvertedFromSuper`
+
+It MUST trigger when anyone converts `SuperchainERC20`
+into legacy tokens with `convertFromSuper`.
+
+```solidity
+event ConvertedFromSuper(address indexed legacyAddr, address indexed superAddr, address indexed caller, uint256 amount);
+```
+
+### Internal Functions
+
+#### `checkPair`
+
+Checks the validity of the `_legacyAddr` and `_superAddr` pair.
+To do so, it will verify:
+
+1. Valid `OptimismMintableERC20` address:
+
+   1. For representations minted before
```

Comment:
I think that we can leave this bullet point for a backwards compatibility section and just say that the validity involves checking the canonical token mapping in the factory + a decimals check. Do we need to do checks for other erc20 metadata equality?

Reply:
I agree it creates some confusion for first time readers. I could move the details to an Appendix. 
No need to check other metadata equality I think.

Reply:
Moved to the Appendix